cmake/HPX_SetupHwloc.cmake

```diff
@@ -13,9 +13,93 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-find_package(Hwloc)
-if(NOT Hwloc_FOUND)
-  hpx_error(
-    "Hwloc could not be found, please specify Hwloc_ROOT to point to the correct location"
+if(NOT HPX_WITH_FETCH_HWLOC)
+  find_package(Hwloc)
+  if(NOT Hwloc_FOUND)
+    hpx_error(
+      "Hwloc could not be found, please specify Hwloc_ROOT to point to the correct location"
+    )
+  endif()
+else()
+  hpx_info(
+    "HPX_WITH_FETCH_HWLOC=${HPX_WITH_FETCH_HWLOC}, Hwloc will be fetched using CMake's FetchContent"
   )
+  if(UNIX)
+    include(FetchContent)
+    fetchcontent_declare(
+      HWLoc
+      URL https://download.open-mpi.org/release/hwloc/v2.9/hwloc-2.9.3.tar.gz
```

Comment:
We may need to keep a separate variable for HWLOC version.

Reply:
I have added them in the latest commit